docs/docsite/rst/dev_guide/developing_locally.rst

```diff
@@ -0,0 +1,95 @@
+* Plugins are shared code that can be used by any module. They provide abilities like cacheing information or copying files that are useful for many modules.
+
+.. _local_modules:
+
+Adding a Module Locally
+=======================
+Ansible automatically loads all executable files found in certain directories as modules, so you can create or add a local module in any of these locations:
+
+* any directory added to the ``ANSIBLE_LIBRARY`` environment variable (``$ANSIBLE_LIBRARY`` takes a colon-separated list like ``$PATH``)
+* ``~/.ansible/plugins/modules/``
+* ``/usr/share/ansible/plugins/modules/``
+
+Once you save your module file in one of these locations, Ansible will load it and you can use it in any local task, playbook, or role.
+
+To confirm that ``my_custom_module`` is available:
+
+* type ``ansible-doc -t module my_custom_module``. You should see the documentation for that module.
+
+To use a local module only in certain playbooks:
+
+* store it in a sub-directory called ``library`` in the directory that contains the playbook(s)
+
+To use a local module only in a single role:
+
+* store it in a sub-directory called ``library`` within that role
+
+.. _distributing_plugins:
+.. _local_plugins:
+
+Adding a Plugin Locally
+=======================
+Ansible loads plugins automatically too, loading each type of plugin separately from a directory named for the type of plugin. Here's the full list of plugin directory names:
+
+    * action_plugins*
+    * cache_plugins
+    * callback_plugins
+    * connection_plugins
+    * filter_plugins*
+    * inventory_plugins
+    * lookup_plugins
+    * shell_plugins
+    * strategy_plugins
+    * test_plugins*
+    * vars_plugins
+
+You can create or add a local plugin in any of these locations:
+
+* any directory added to the relevant ``ANSIBLE_plugin_type_PLUGINS`` environment variable (these variables, such as ``$ANSIBLE_INVENTORY_PLUGINS`` and ``$ANSIBLE_VARS_PLUGINS`` take colon-separated lists like ``$PATH``)
+* the directory named for the correct ``plugin_type`` within ``~/.ansible/plugins/`` - for example, ``~/.ansible/plugins/callback_plugins``
+* the directory named for the correct ``plugin_type`` within ``/usr/share/ansible/plugins/`` - for example, ``/usr/share/ansible/plugins/plugin_type/action_plugins``
+
+Once your plugin file is in one of these locations, Ansible will load it and you can use it in a any local module, task, playbook, or role.
+
+To confirm that ``plugins/plugin_type/my_custom_plugin`` is available:
+
+* type ``ansible-doc -t <plugin_type> my_custom_lookup_plugin`` - for example, ``ansible-doc -t lookup my_custom_lookup_plugin``. You should see the documentation for that plugin. This works for all plugin types except the ones marked with ``*`` in the list above  - see :ref:`ansible-doc` for more details.
```

Comment:
Change '- for example' to use an em-dash: `--- for example`, which would render ' — for example' (or just insert a an actual unicode em-dash)